querybook/webapp/ui/Comment/Reactions.tsx

```diff
@@ -29,34 +29,27 @@ export const Reactions: React.FunctionComponent<IProps> = ({
         [commentId, dispatch]
     );
     const deleteEmoji = React.useCallback(
-        (emoji: string, uid: number) => {
-            const reactionToDelete = uidsByReaction[uid]?.find(
-                (reaction) =>
-                    reaction.reaction === emoji && reaction.created_by === uid
-            );
-            if (reactionToDelete) {
-                dispatch(
-                    deleteReactionByCommentId(commentId, reactionToDelete.id)
-                );
-            }
+        (reactionId) => {
+            dispatch(deleteReactionByCommentId(commentId, reactionId));
         },
-        [uidsByReaction, dispatch, commentId]
+        [dispatch, commentId]
     );
 
-    const handleReactionClick = (reaction: string, uid: number) => {
-        const uidIdx = uidsByReaction[reaction].findIndex(
-            (uid) => uid === userInfo.uid
+    const handleReactionClick = (emoji: string, uid: number) => {
+        const existingReaction = reactionsByEmoji[emoji].find(
+            (reaction) =>
+                reaction.reaction === emoji && reaction.created_by === uid
```

Comment:
as it's already reactions by emoji, unnecessary to check it again `reaction.reaction === emoji`